Gems/SimulationInterfaces/Code/Source/Clients/NamedPosesManager.cpp

```diff
@@ -0,0 +1,291 @@
+            if (parentSimulatedBody == nullptr)
+            {
+                auto msg = AZStd::string::format("Entity's simulated body doesn't exist");
+                return AZ::Failure(msg);
+            }
+            if (parentSimulatedBody->m_bodyHandle == AzPhysics::InvalidSimulatedBodyHandle)
+            {
+                auto msg = AZStd::string::format("Entity is not valid simulated body");
+                return AZ::Failure(msg);
+            }
+            return AZ::Success(parentSimulatedBody);
+        }
+    } // namespace
+
+    AZ_COMPONENT_IMPL(NamedPoseManager, "NamedPoseManager", NamedPoseManagerTypeId);
+
+    void NamedPoseManager::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<NamedPoseManager, AZ::Component>()->Version(0);
+        }
+    }
+
+    void NamedPoseManager::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+    {
+        provided.push_back(AZ_CRC_CE("NamedPoseManagerService"));
+    }
+
+    void NamedPoseManager::GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+    {
+        incompatible.push_back(AZ_CRC_CE("NamedPoseManagerService"));
+    }
+
+    void NamedPoseManager::GetRequiredServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& required)
+    {
+        required.push_back(AZ_CRC_CE("SimulationFeaturesAggregator"));
+    }
+
+    void NamedPoseManager::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    {
+        dependent.push_back(AZ_CRC_CE("PhysicsService"));
+        dependent.push_back(AZ_CRC_CE("SimulationFeaturesAggregator"));
+    }
+
+    NamedPoseManager::NamedPoseManager()
+    {
+        if (NamedPoseManagerRequestBusInterface::Get() == nullptr)
+        {
+            NamedPoseManagerRequestBusInterface::Register(this);
+        }
+    }
+
+    NamedPoseManager::~NamedPoseManager()
+    {
+        if (NamedPoseManagerRequestBusInterface::Get() == this)
+        {
+            NamedPoseManagerRequestBusInterface::Unregister(this);
+        }
+    }
+
+    void NamedPoseManager::Init()
+    {
+    }
+
+    void NamedPoseManager::Activate()
+    {
+        SimulationFeaturesAggregatorRequestBus::Broadcast(
+            &SimulationFeaturesAggregatorRequests::AddSimulationFeatures,
+            AZStd::unordered_set<SimulationFeatureType>{
+                simulation_interfaces::msg::SimulatorFeatures::NAMED_POSES,
+                simulation_interfaces::msg::SimulatorFeatures::POSE_BOUNDS,
+            });
+
+        NamedPoseManagerRequestBus::Handler::BusConnect();
+    }
+
+    void NamedPoseManager::Deactivate()
+    {
+        NamedPoseManagerRequestBus::Handler::BusDisconnect();
+    }
+
+    AZ::Outcome<void, FailedResult> NamedPoseManager::RegisterNamedPose(AZ::EntityId namedPoseEntityId)
```

Comment:
```suggestion
    AZ::Outcome<void, FailedResult> NamedPoseManager::RegisterNamedPose(AZ::EntityId& namedPoseEntityId)
```
Is there a reason to do not use a reference?